design/user-id-update.md

```diff
@@ -0,0 +1,47 @@
+## Make User IDs persistent across session ID, but unique per AppID
+
+This is related to github issue #129.
+
+I want us to redo user IDs and persistence. Current implementation is inconsistent across testbed+Altspace, is inconsistent across registered/unregistered apps, and was designed this way based off of us not having a permission system.
+
+For Background for current implementation, see Issue #129:
+1. Eric wrote: "In a space where an MRE is instantiated multiple times with the same sessionId, a given user in the world will have a different userId in each instance."
+2. Steven wrote this comment : "@eanders-MS This is by design. GDPR requires us to not allow users to be tracked without their permission. As such, any given Altspace user will be assigned a different ID for each instance of a given app. I agree that a persistent ID per app would be tremendously useful, but we have to come up with a strategy to protect our users' privacy first."
+3. Eric wrote about a design:
+	"Registered apps: Same userId for a given user across multiple instances of an app (because the AppId is consistent).
+	Unregistered apps: Different userIds for a given user across multiple instances of an app (because the AppId is derived from the space component id)."
+	"We talked about adding functionality on the website for users to be able to be forgotten by all apps. How this would work: When generating the userId, include a key taken from the user record. When the user chooses to be forgotten, we would roll this key"
+
+Currently in Altspace we use: UserID= combination of altspaceUserID and either appid (for registered apps) or SCID
+
+Currently in testbed we use: UserID=Random generated every connection
+
+### Proposed design
+1. For all host apps have user runtime hash a persistent userID with an AppID or (if AppID isn't available) URL. 
+2. Have the hashing logic be part of the unity runtime instead of be the host app's responsibility
+3. Provide a workaround that allows multi-instance testing as a single user, by adding an optional postfix to the session ID, which gets hash with the User ID by the client
```

Comment:
>session ID [](start = 114, length = 10)

As mentioned in Background, the session ID does not contribute to the user ID. Did you mean app ID?

Reply:
correct, this has been cleaned up in updated design